### PR TITLE
Use json format to save and load partition config

### PR DESCRIPTION
**Pull request by RandySheriffH**
Use json format to save and load partition config, previously it was csv, which brought issues among windows and posix due to different line breaks.

### CODE REVIEW

File: onnxruntime/core/framework/allocation_planner.cc [hunk 1]
```diff
@@ -2207,160 +2213,74 @@ Status SequentialPlanner::CreatePlan(
...
-  }
+
+  bool DumpPartition() const;
+
+  Status PartitionGraph(const onnxruntime::GraphViewer& graph_viewer,
+                        const ExecutionProviders& execution_providers,
+                        std::vector<InlinedVector<NodeIndex>>& stream_nodes,
+                        ExecutionOrder execution_order) override;
+
+  const char* Type() const override { return "DeviceBasedPartitioner"; }
```
Comment (jslhcl) [inline onnxruntime/core/framework/allocation_planner.cc:2238]: nit: use enum for Type?

File: onnxruntime/core/framework/allocation_planner.h [hunk 2]
```diff
@@ -78,33 +79,29 @@ class SequentialPlannerContext : public ISequentialPlannerContext {
...
-  // i.e., given a graph which has CPU EP nodes, Cuda EP nodes and TRT EP nodes,
-  // it will be partitioned as two sequences, one is for CPU EP nodes, another is for TRT and Cuda nodes.
-  // We will add more optimized partitioner later.
   enum GraphPartitioningStrategy {
     DeviceBasedPartition = 0,
     Unknown,
   };
+  // DeviceBasedPartition is the default, who partitions a graph based off device information.
+  // i.e., given a graph which has CPU EP nodes, Cuda EP nodes and TRT EP nodes,
+  // it will be partitioned as two sequences, one is for CPU EP nodes, another is for TRT and Cuda nodes.
```
Comment (jslhcl) [inline onnxruntime/core/framework/allocation_planner.h:88]: shouldn't be 3 sequences for this case? 1 for cpu, 1 for TRT, 1 for Cuda?

File: onnxruntime/test/testdata/multi_stream_models/simplified_ssd_cpu.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+"type":"DeviceBasedPartitioner",
+"streams":[["conv_0","relu_0","maxpool_0","conv_4","add_0"],["conv_1","relu_1","conv_2","relu_2","conv_3"]]
+}
```
Comment (jslhcl) [inline onnxruntime/test/testdata/multi_stream_models/simplified_ssd_cpu.json:4]: Can we document the JSON format somewhere?

File: onnxruntime/core/framework/allocation_planner.h (hunk 2 above)
Comment (RandySheriffH) [reply onnxruntime/core/framework/allocation_planner.h]: TRT and Cuda EP nodes should all fall into Cuda device category.

File: onnxruntime/core/framework/allocation_planner.cc (hunk 1 above)
Comment (RandySheriffH) [reply onnxruntime/core/framework/allocation_planner.cc:2235]: Enum is for sure better - planning on adding more partitioners using different algorithms, will extend to Enum shortly on that change.

File: onnxruntime/test/testdata/multi_stream_models/simplified_ssd_cpu.json (hunk 3 above)
Comment (RandySheriffH) [reply onnxruntime/test/testdata/multi_stream_models/simplified_ssd_cpu.json:4]: Exactly, done.

File: onnxruntime/core/framework/allocation_planner.cc [hunk 4]
```diff
@@ -1748,8 +1753,9 @@ class PlannerImpl {
 
   void
   PartitionIntoStreams(const logging::Logger& logger, const ExecutionProviders& execution_providers,
-                       const std::string& partition_config_file) {
+                       const std::basic_string<PATH_CHAR_TYPE>& partition_config_file) {
     auto partitioner = IGraphPartitioner::CreateGraphPartitioner(logger, partition_config_file);
```
[the remark this answers is not in the document]
Comment (RandySheriffH) [reply onnxruntime/core/framework/allocation_planner.cc:1757]: Done.

**Review by souptc — APPROVED** [review]

**Pull request merged**